change wallet categories into wallet quadrant

8go opened the pull request:
The original categorization was not perfect. It was not ideal. It was sort of misleading. It sort of implies that a "Full Node & Wallet" is not a "Non-Custodial Wallet" because they are 2 different categories. It also states that a "Non-Custodial Wallet" uses "3rd-party node" for LN Node which is not the true definition.Things were getting mixed up here and we are not communicating a clear message.

The term "Custodial Wallet" is independent of which LN Node is used.
We should rethink this table and categorization.

I wrote an Issue about it #176 .

The Quadrant seems to be more useful and certainly more accurate.

Then I explained the terms "Lightweight" and "Node" because these terms were suddenly used, out of nowhere with no explanation.

Commit:
change wallet categories into wallet quadrant

The original categorization was not perfect. It was not ideal. It was sort of misleading. It sort of implies that a "Full Node & Wallet" is not a "Non-Custodial Wallet" because they are 2 different categories. It also states that a "Non-Custodial Wallet" uses "3rd-party node" for LN Node which is not the true definition.Things were getting mixed up here and we are not communicating a clear message.

The term "Custodial Wallet" is independent of which LN Node is used.
We should rethink this table and categorization.

I wrote an Issue about it #176 .

The Quadrant seems to be more useful and certainly more accurate.

Then I explained the terms "Lightweight" and "Node" because these terms were suddenly used, out of nowhere with no explanation.

Roasbeef commented on 02_getting_started.asciidoc:
```diff
@@ -36,33 +36,48 @@ If users decide to outsource some functionality to a third party they usually gi
 
-In <<lnwallet-categories>> we see the three broad categories of lightning wallets and the relative degree of control they offer to the user.
+There are many ways wallets can be characterized or categorized. 
+From all features the two most important criteria are LN Node use and fund custody. Correspondingly the two most important questions to ask about a specific wallet are:
```
I think a third question is something along the lines of where it gets its Bitcoin blockchain data from. Even if you have a "full" LN wallet, and control your own keys, if you blindly accept bogus data from a 3rd party, then you can actually be using "fake coins" without knowing it.

8go replied:
Added the question. See Commit # 2  72314dab8d709783af287d97600e712b9dc91a06

Roasbeef commented on 02_getting_started.asciidoc:
```diff
@@ -36,33 +36,48 @@ If users decide to outsource some functionality to a third party they usually gi
 
 [[lnwallet-categories]]
-.Broad Categories of "Lightning Wallets"
+.Lightning Wallets Quadrant
```
I dig this matrix idea! I think we should add another row for the items I mentioned in #213.

8go replied:
reworked Quadrant and text,
But I still kept it as a Quadrant. 
As high-level as possible-> 4 quadrants

See 72314dab8d709783af287d97600e712b9dc91a06

Roasbeef commented on 02_getting_started.asciidoc:
```diff
@@ -36,33 +36,48 @@ If users decide to outsource some functionality to a third party they usually gi
+| *Custodial*            | Q3: not a common wallet type | Q4: Low technical skills 
 |===
 
+Q3, quadrant 3, where a full LN node is used but the keys are held by a custodian is not common. A user who goes through the effort of running a Ligtning Node will most likely also put in the additional efffort to store his own keys. 
```
True that it's not common, but it may become attractive in the future, as it would let a user worry about the operational aspects of their node, but then possibly delegate the access to the keys+signing to a 3rd party which may use primarily cold storage.

8go replied:
I rephrased and worked your idea into the text. See Commit # 2.

See 72314dab8d709783af287d97600e712b9dc91a06

Roasbeef commented on 02_getting_started.asciidoc:
```diff
@@ -36,33 +36,48 @@ If users decide to outsource some functionality to a third party they usually gi
 
+The category "Third-party Ligntning Nodes" can again be subdivided into:
+
+- Lightweight: means that the Lightning Node is operated by a third party and that the wallet obtains the required information through an API that connects wallet and third-party LN node.
```
Hmm, I think we're getting close here, but there's still more nuance. One example is the new Phoenix wallet. The user controls the keys, grabs blockchain data from Electrum servers (I believe, making them SPV in a sense), but delegates stuff like opening+managing channels and path finding to a 3rd party server. 

It sits somewhere in the middle, making certain tradeoffs to provide a better user experience in a mobile setting. 

Something on the far opposite end is something like Bluewallet, or other fully custodial wallets. In this case, the user has minimal control and trusts the operator of the hub node to not run away with funds. In the example of the Phoenix wallet, the user still maintains control of their keys.

8go replied:
How is this not captured accurately in the table?
Phoenix Wallet | Mobile  | LN=Lightweight | BTC=Electrum | Self-Custody

Let's look at your sentence:
Phoenix wallet. The user controls the keys,  --> Self-custody

grabs blockchain data from Electrum servers (I believe, making them SPV in a sense),  --> Electrum

 but delegates stuff like opening+managing channels and path finding to a 3rd party server --> LN=3rd-party Lightweight

For Phoenix: As far as I can see your sentence matches exactly the table. What is it that I am not seeing?
Same for Blue Wallet. 
Unless I am missing something, it is OK as is. Please give further details to clarify. Thanks.

Roasbeef commented on 02_getting_started.asciidoc:
```diff
@@ -36,33 +36,48 @@ If users decide to outsource some functionality to a third party they usually gi
+| lnd            | Server  | Full Node   | Bitcoin Core/btcd     | Self-Custody
+| c-lightning    | Server  | Full Node   | Bitcoin Core          | Self-Custody
+| Eclair Server  | Server  | Full Node   | Bitcoin Core/Electrum | Self-Custody
+| Zap Desktop    | Desktop | Full Node   | Bitcoin Core/btcd     | Self-Custody
```
For both Zap and Breez, we might want to add "neutrino" instead as the Bitcoin node category.

8go replied:
Currently:
| Zap Desktop    | Desktop | Full Node   | Bitcoin Core/btcd     | Self-Custody

I agree, this creates the wrong implication. It could lead to the incorrect conclusion that Zap Wallet runs a Bitcoin Core Full node. Not so!
This needs to be fixed. Thanks for catching that and pointing it out!

I fix Zap and Breez in Commit # 2.  72314dab8d709783af287d97600e712b9dc91a06

Roasbeef reviewed (changes requested):
Thanks for this PR! I agree we can improve the way we describe the various types of wallets, as there's a lot of nuance there, and as time goes on, developers are creating even newer models. I really dig the introduction of this matrix/quadrant system, as we can extend it in the future, and it gives readers a peek into all the combinations (and innovation!) that's possible.

Commit:
various improvements

- Following suggestions from  @Roasbeef 
- worked in his commentaries, ideas
- also added a bit more details

8go commented:
Look at 72314dab8d709783af287d97600e712b9dc91a06

I worked all comments in. I think the PR can be pulled.

The chapter has to be improved further over time. Many wallets will need to be added ....

This PR at least sets the structure and terms to use in table.

Roasbeef commented on 02_getting_started.asciidoc:
```diff
@@ -36,33 +36,59 @@ If users decide to outsource some functionality to a third party they usually gi
+
+Other terms that need explanation in Table <<lnwallet-examples>> in column "Bitcoin Node" are:
+
+- Neutrino: Neutrino is a protocol to get data from Bitcoin. The Bitcoin Node is run by a third-party and accessed via "neutrino".
```
Can point to BIP 157+158. Neutrino is the name of a _specific_ implementation of the client-side/wallet portion of these BIPs. However overtime, people have come to colloquially refer an implementation of the BIPs as "neutrino".

8go replied:
Good idea, that makes is better.

Done, see Commit # 3 in 6d3e4e37409994a2ebd2e1f35c53196a32d6ca4f

Roasbeef reviewed (changes requested):
One final correctness comment. Agree that this is just a start and will evolve a ton over time.

Commit:
added BIPs to Neutrino

renepickhardt committed:
Merge branch 'develop' into patch-37

renepickhardt commented:
Thanks a lot!
